problems/[230615] Symmetric Tree.js

```diff
@@ -0,0 +1,39 @@
+/**
+ * leetcode problem link: https://leetcode.com/problems/symmetric-tree
+ *
+ * Definition for a binary tree node.
+ * function TreeNode(val, left, right) {
+ *     this.val = (val===undefined ? 0 : val)
+ *     this.left = (left===undefined ? null : left)
+ *     this.right = (right===undefined ? null : right)
+ * }
+ */
+/**
+ * @param {TreeNode} root
+ * @return {boolean}
+ */
+const isSymmetric = function (root) {
+  if (!root) {
+    return false;
+  }
+
+  return checkIsSame(root.left, root.right);
+
+  function checkIsSame(leftNode, rightNode) {
+    if (
+      (!leftNode && rightNode) ||
+      (leftNode && !rightNode) ||
+      (leftNode && rightNode && leftNode.val !== rightNode.val)
+    ) {
+      return false;
+    }
+
+    if (leftNode && rightNode) {
+      return (
+        checkIsSame(leftNode.left, rightNode.right) &&
+        checkIsSame(leftNode.right, rightNode.left)
+      );
+    }
+    return true;
+  }
```

Comment:
```suggestion
   if (leftSub == null && rightSub == null) return true;

   return (leftNode && rightNode) 
        &&  checkIsSame(leftNode.left, rightNode.right)
        &&  checkIsSame(leftNode.right, rightNode.left);
  }
```

여기를 이렇게 줄 일 수 있긴한데, 그럼 true 조건을 위에서 한 번 더 걸러야겠네요!